sys/androidmedia/gstamcvideodec.c

```diff
@@ -1565,9 +1565,19 @@ gst_amc_video_dec_set_format (GstVideoDecoder * decoder,
       (is_format_change || (is_size_change && !adaptive));
   needs_config = !self->started || needs_disable;
 
-  /* FIXME: This is not an error, but we want to force this log
-   * and for now we can only achieve it in android using GST_ERROR */
-  GST_ERROR_OBJECT (self, "needs_disable=%d needs_config=%d", needs_disable,
+  if (self->drm_ctx) {
+    gboolean need_drm_reconfigure;
+
+    if (G_UNLIKELY (!gst_amc_drm_mcrypto_update (self->drm_ctx,
+                &need_drm_reconfigure))) {
+      GST_ERROR_OBJECT (self, "Failed to update MediaCrypto");
+      return FALSE;
+    }
+    needs_config |= need_drm_reconfigure;
+    needs_disable |= need_drm_reconfigure;
+  }
+
+  GST_INFO_OBJECT (self, "needs_disable=%d needs_config=%d", needs_disable,
```

Comment:
Check this trace. Different level than above, we may not know whats the final state. You may use level error for both, or leave only the last one but with level error and add a trace with need_drm_reconfigure.

Reply:
This is not an error message, but an info message. Error loglevel was there only because in the past it was a way to make log appear in fluendo-sdk without modifications. But now flusdk has line:

  gst_debug_set_threshold_for_name ("*amc*", GST_LEVEL_INFO);

that makes info log in code of amc codecs appear, as well as this one. I have checked that, this log appears in adb logcat:

`08-13 18:53:46.754  6221  6426 I GStreamer+amcvideodec: 0:00:02.867116168 0x865d18f0 gstamcvideodec.c:1581:gst_amc_video_dec_set_format:<amcvideodec-omxhisivideodecoderavc1> needs_disable=0 needs_config=1`